src/constants.ts

```diff
@@ -0,0 +1,34 @@
+import { isExactNumber } from './util'
+
+export const IPN_TYPES_ENUM = {
```

Comment:
why not use actual enum type?

Reply:
Because of how the type computes. For some reason `enum` does not get computed when used with other types. `as const` plays well in that manner.